src/components/UserForm/BasicInformationSection.js

```diff
@@ -52,6 +72,13 @@ const BasicInformationSection = React.memo(
 
         return (
             <FormSection title={i18n.t('Basic information')}>
+                {enforceLoading && (
```

Comment:
I'm not sure the loader is really necessary and removing it would have allowed you to collocate the query in for the `enforceVerifiedEmail` setting in the EmailStatusMessage component, which I think would have maybe improved code readability. 

With this loader (try with 3G connection), I believe there's a brief state where useDataQuery returns {loading:false, data:undefined} which can lead to the text message flickering from grey to red (when user does not have verified email and enforceVerifiedEmail:true).

If you had the query in the EmailStatusMessage component you'd have flickering from not showing the message to showing the message, but I'm not sure that's any worse 🤔

I would probably vote to not change it at this point.